packages/aws-cdk-lib/aws-apigateway/README.md

```diff
@@ -465,7 +465,7 @@ declare const api: apigateway.RestApi;
 
 const key = new apigateway.RateLimitedApiKey(this, 'rate-limited-api-key', {
   customerId: 'hello-customer',
-  stages: [api.deploymentStage],
```

Comment:
It seems that the RateLimitApiKey has a [stages](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway.RateLimitedApiKey.html#stages) property.
Can you tell me more about the meaning of 'not available'?

Reply:
@mazyu36  When I used the `stages`, I observed that while the api key was associated with the provisioned Usage Plan. But the Stage was not. This is what I meant by "not available". ( I realizing this expression was not good...)

If we were to follow the sample code using `stages`, it would be necessary to call `plan.addApiStage`. That means would require declaring the `UsagePlan` explicitly.
In such a case, since we would be setting the rate limit in declare `UsagePlan`, the necessity of `RateLimitedApiKey` becomes unclear.

Therefore, I read #6509 to understand the context behind the implementation of `RateLimitedApiKey`.
I gathered that it was implemented to declare rate limit related configure and associated with usage plan and stage, not only declaring api key.
if current behavior and my understanding is correct, I think sample code is suitable for using `apiStages` instead of `stages` 

Could you please advice me when my understand is wrong?

Reply:
@akagire 
Thank you for the explanation.
I see that `apiStages` are indeed linked to the `UsagePlan` as you described. 
https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-apigateway/lib/api-key.ts#L261

Since this modification aligns the code example with a more appropriate use case, I think it's fine. 
Let's see what the maintainers think about it.

Could you please update the issue description? 
This is because the `stages` property itself is not actually unavailable.

Reply:
@akagire thanks for taking on this doc update! it seems confusing to me that `apiStages` and `stages` are both available on the construct but are applied in different areas. if that's not a bug, can we also add in the README a more clear distinction for how those properties differ?

Reply:
@mazyu36 I updated.

@kaizencc Honestly, I cannot describe those distinction yet but I'll try to and investigate. The possibility of a bug has not been ruled out. If so, I'll post new issue.

Reply:
this still seems like an open thread that should be closed before this PR is merged